core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -658,6 +628,37 @@ public static int lastIndexOfAscii(@NotNull Utf8Sequence seq, char asciiTerm) {
         return -1;
     }
 
+    /**
+     * Strictly less than (&lt;) comparison of two UTF8 sequences in lexicographical
+     * order. For example, for:
+     * l = aaaaa
+     * r = aaaaaaa
+     * the l &lt; r will produce "true", however for:
+     * l = bbbb
+     * r = aaaaaaa
+     * the l &lt; r will produce "false", because b &lt; a.
+     *
+     * @param l left sequence, can be null
+     * @param r right sequence, can be null
+     * @return if either l or r is "null", the return value false, otherwise sequences are compared lexicographically.
+     */
+    public static boolean lessThan(@Nullable Utf8Sequence l, @Nullable Utf8Sequence r) {
+        if (l == null || r == null) {
```

Comment:
How `lessThan` is used? This definition is not anti-symmetric - so it violates [total ordering](https://en.wikipedia.org/wiki/Total_order) rules. If this comparator is used for some sorting routine - it can lead to unpredictable behaviour in case when `null`s are in the sequence.

Reply:
This is used in SQL `l < r` expression, and aligns with SQL semantics that say such an expression is always false if one of the operands is unknown, i.e., NULL.